src/formats/anvil/anvil_world.py

```diff
@@ -214,25 +214,22 @@ def get_blocks(self, cx: int, cz: int) -> Union[numpy.ndarray, NotImplementedErr
             internal = self._materials.get_block_from_definition(
                 block, default="minecraft:unknown_{}"
             )
+            block_object: Block = self.get_blockstate(internal)
             if (
                 internal == "minecraft:unknown_{}"
             ):  # If we don't have the block in our definitions, call it an unknown block
                 try:
                     internal_id = list(self.unknown_blocks.values()).index(block)
                 except ValueError:
-                    internal_id = len(self.mapping_handler)
-                    internal = internal.format(len(self.unknown_blocks))
+                    block_object: Block = Block(
+                        blockstate=f"minecraft:unknown_{len(self.unknown_blocks)}"
+                    )
+                    internal_id = self.mapping_handler[block_object]
```

Comment:
It's really implicit here that it first sets the block_object into mapping_handler and then finds its location. The part that adds block_object into mapping_handler should explicit and probably on a separate line. So it will first add it to mapping_handler and only then find the internal_id.

Reply:
With the way BlockManager is currently implemented, that method won't really work since that's how BlockManager handles adding new Block objects, maybe add a comment that explains what's happening?

Reply:
Adding a comment does not help. I prefer the method of "self-documenting code" where the code is enough to understand what's going on. If the BlockManager class doesn't allow us to change it, we should change the BlockManager class to fit our needs. Can you explain the issue with the current implementation of BlockManager?